astroid/objects.py

```diff
@@ -266,11 +266,13 @@ class DictValues(bases.Proxy):
 class PartialFunction(scoped_nodes.FunctionDef):
     """A class representing partial function obtained via functools.partial"""
 
+    @decorators.deprecate_arguments(doc="Use the postinit arg 'doc_node' instead")
     def __init__(
         self, call, name=None, doc=None, lineno=None, col_offset=None, parent=None
     ):
         # TODO: Pass end_lineno and end_col_offset as well
-        super().__init__(name, doc, lineno, col_offset, parent=None)
+        super().__init__(name, lineno=lineno, col_offset=col_offset, parent=None)
+        self._doc = doc
```

Comment:
Without this we raise the `DeprecationWarning` from the `__init__` of `FunctionDef`. This avoid this.

Reply:
Should we add a small comment?
```
Assign doc directly to prevent DeprecationWarning
```

Reply:
Should we add a small comment?
```
Assign doc directly to prevent DeprecationWarning
```

Reply:
I'll add one and a TODO